src/dmd/root/filename.d

```diff
@@ -975,14 +975,15 @@ nothrow:
                 char[PATH_MAX] buf = void;
                 auto path = name.toCStringThen!((n) => realpath(n.ptr, buf.ptr));
                 if (path !is null)
-                    return mem.xstrdup(path).toDString;
+                    return xarraydup(path.toDString);
             }
             else static if (__traits(compiles, canonicalize_file_name))
             {
                 // Have canonicalize_file_name, which malloc's memory.
                 auto path = name.toCStringThen!((n) => canonicalize_file_name(n.ptr));
+                scope(exit) .free(path.ptr);
                 if (path !is null)
-                    return path.toDString;
+                    return xarraydup(path.toDString);
```

Comment:
Can you add a comment, e.g. `// Make sure we consistently return `Mem`-allocated memory`. Otherwise someone is bound to see this and think it can be trivially improved.

Reply:
To prevent such an 'improvement', I've added the `FileName.free()` call in the test. But that needs an enabled GC (`-lowmem`) to hopefully produce an error.